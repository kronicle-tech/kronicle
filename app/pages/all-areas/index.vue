<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllAreasTabs />

          <b-alert show="10" dismissible variant="info" class="my-3">
            Click an area's name in the table below to view more information about
            that area
          </b-alert>

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
              <tr>
                <th class="area-name">Name</th>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr v-for="area in areas" :key="area.id">
                <td class="area-name table-primary">
                  <AreaName :area="area" />
                </td>
                <td>
                  <Markdown :markdown="area.description" />
                </td>
              </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BAlert, BCol, BContainer, BRow} from 'bootstrap-vue'
import { Area } from '~/types/kronicle-service'
import AllAreasTabs from '~/components/AllAreasTabs.vue'
import AreaName from '~/components/AreaName.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllAreasTabs,
    'b-alert': BAlert,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    AreaName,
    Markdown,
  },
  async asyncData({ $config }) {
    const areas = await fetch(
      `${$config.serviceBaseUrl}/v1/areas?fields=areas(id,name,description)`
    )
      .then((res) => res.json())
      .then((json) => json.areas as Area[])

    return {
      areas,
    }
  },
  data() {
    return {
      areas: [] as Area[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Areas',
    }
  },
})
</script>
