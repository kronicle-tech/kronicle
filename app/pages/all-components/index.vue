<template>
  <div>
    <AllComponentsTabs />
    <b-alert show="10" dismissible variant="info" class="my-3">
      Click a component's name in the table below to view more information about
      that component
    </b-alert>

    <ComponentsView :components="components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BAlert } from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Component } from '~/types/kronicle-service'
import AllComponentsTabs from '~/components/AllComponentsTabs.vue'
import ComponentsView from '~/components/ComponentsView.vue'

export default Vue.extend({
  components: {
    AllComponentsTabs,
    'b-alert': BAlert,
    ComponentsView,
  },
  async asyncData({ $config, route, store }) {
    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId)`
    )
      .then((res) => res.json())
      .then((json) => json.components as Component[])

    store.commit('componentFilters/initialize', {
      components,
      route,
    })

    return {
      components,
    }
  },
  data() {
    return {
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Components',
    }
  },
})
</script>
