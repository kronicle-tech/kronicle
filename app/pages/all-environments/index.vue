<template>
  <div class="m-3">
    <EnvironmentStatesView :components="components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {NuxtRuntimeConfig} from "@nuxt/types/config/runtime";
import {Route} from "vue-router";
import {Store} from "vuex";
import EnvironmentStatesView from '~/components/EnvironmentStatesView.vue'
import { Component } from '~/types/kronicle-service'

interface Data {
  components: Component[]
}

async function loadData(
  { $config, route, store }: { $config: NuxtRuntimeConfig, route: Route, store: Store<any> }
): Promise<Data> {
  const components = await fetch(
    `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId,states)`
  )
    .then((res) => res.json())
    .then((json) => json.components as Component[])

  store.commit('componentFilters/initialize', {
    components,
    route,
  })

  return {
    components: [] as Component[],
  }
}

export default Vue.extend({
  components: {
    EnvironmentStatesView,
  },
  async asyncData({ $config, route, store }) {
    return await loadData({
      $config,
      route,
      store,
    })
  },
  data() {
    return {
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Environments',
    }
  },
  mounted() {
    setInterval(
      async () => {
        const data = await loadData({
          $config: this.$config,
          route: this.$route,
          store: this.$store
        })
        this.components = data.components
      },
      60 * 1000
    )
  },
})
</script>
