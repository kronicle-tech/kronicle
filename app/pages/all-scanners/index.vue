<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllScannersTabs />

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
            <tr>
              <th class="scanner-id">Id</th>
              <th class="scanner-description">Description</th>
              <th class="scanner-notes">Notes</th>
            </tr>
            </thead>
            <tbody>
            <tr v-for="scanner in scanners" :key="scanner.id">
              <td class="scanner-id">
                {{ scanner.id }}
              </td>
              <td class="scanner-description">
                <Markdown :markdown="scanner.description" />
              </td>
              <td class="scanner-notes">
                <Markdown :markdown="scanner.notes" />
              </td>
            </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BCol, BContainer, BRow} from "bootstrap-vue";
import { Scanner } from '~/types/kronicle-service'
import AllScannersTabs from '~/components/AllScannersTabs.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllScannersTabs,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    Markdown,
  },
  async asyncData({ $config }) {
    const scanners = await fetch(
      `${$config.serviceBaseUrl}/v1/scanners?fields=scanners(id,description,notes)`
    )
      .then((res) => res.json())
      .then((json) => json.scanners as Scanner[])

    return {
      scanners,
    }
  },
  data() {
    return {
      scanners: [] as Scanner[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Scanners',
    }
  },
})
</script>
