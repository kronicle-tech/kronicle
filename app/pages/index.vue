<template>
  <div class="m-3">
    <Intro id="intro" :title="introTitle" :markdown="introMarkdown" />

    <b-card-group columns>
      <b-card
        header="Kronicle"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Kronicle contains the following:
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-areas' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Areas</b>
            <span class="lead">
              <b-badge class="area-count" variant="primary" pill>{{
                  areaCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-teams' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Teams</b>
            <span class="lead">
              <b-badge class="team-count" variant="primary" pill>{{
                  teamCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Components</b>
            <span class="lead">
              <b-badge class="component-count" variant="primary" pill>{{
                  componentCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components-tech-debts' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Tech Debts</b>
            <span class="lead">
              <b-badge class="tech-debt-count" variant="primary" pill>{{
                  techDebtCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components-tests' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Test Results</b>
            <span class="lead">
              <b-badge class="test-result-count" variant="primary" pill>{{
                  testResultCount
                }}</b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Scanners"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Scanners dynamically add data to Kronicle at runtime:
          </b-list-group-item>

          <b-list-group-item v-for="scanner in scanners" :key="scanner.id">
            <b class="scanner-id text-info">{{ scanner.id }}</b>
            <br />
            <Markdown :markdown="scanner.description" class="description" />
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Tests"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Tests evaluate the data associated with components in Kronicle and
            return outcomes of pass, fail and not applicable
          </b-list-group-item>

          <b-list-group-item
            v-for="test in tests"
            :key="test.id"
            :to="{
              name: 'tests-testId',
              params: { testId: test.id },
            }"
          >
            <b class="test-id text-info">{{ test.id }}</b>
            <br />
            <Markdown :markdown="test.description" class="description" />
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Platforms"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group-item
          class="d-flex justify-content-between align-items-center text-muted"
        >
          The following platforms are associated with components in Kronicle:
        </b-list-group-item>

        <b-list-group flush>
          <b-list-group-item
            v-for="platformCount in platformCounts"
            :key="platformCount.item"
            :to="{
              name: 'all-components',
              query: { platformId: platformCount.item || 'undefined' },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="platform">{{ platformCount.item || 'missing' }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ platformCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Tags"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            The components in Kronicle are tagged with the following:
          </b-list-group-item>

          <b-list-group-item
            v-for="tagCount in tagCounts"
            :key="tagCount.item"
            :to="{
              name: 'all-components',
              query: { tag: tagCount.item },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="tag">{{ tagCount.item }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ tagCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Component Types"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            There are the following types of components in Kronicle:
          </b-list-group-item>

          <b-list-group-item
            v-for="componentTypeCount in componentTypeCounts"
            :key="componentTypeCount.item"
            :to="{
              name: 'all-components',
              query: { componentType: componentTypeCount.item },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="component-type">{{ componentTypeCount.item }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ componentTypeCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>
    </b-card-group>
  </div>
</template>

<style scoped>
.description {
  display: inline-block;
  color: white;
  text-decoration: none;
}

.description >>> p {
  margin-bottom: 0;
}
</style>

<script lang="ts">
import Vue from 'vue'
import {
  BBadge,
  BCard,
  BCardGroup,
  BListGroup,
  BListGroupItem,
} from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Area, Component, Scanner, Team, Test } from '~/types/kronicle-service'
import { ItemCount, itemCounts } from '~/src/arrayUtils'
import Intro from '~/components/Intro.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    'b-badge': BBadge,
    'b-card': BCard,
    'b-card-group': BCardGroup,
    'b-list-group': BListGroup,
    'b-list-group-item': BListGroupItem,
    Intro,
    Markdown,
  },
  async asyncData({ $config }) {
    const areas = await fetch(
      `${$config.serviceBaseUrl}/v1/areas?fields=areas(id)`
    )
      .then((res) => res.json())
      .then((json) => json.areas as Area[])

    const teams = await fetch(
      `${$config.serviceBaseUrl}/v1/teams?fields=teams(id)`
    )
      .then((res) => res.json())
      .then((json) => json.teams as Team[])

    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,typeId,platformId,tags,techDebts(doesNotExist),testResults(doesNotExist))`
    )
      .then((res) => res.json())
      .then((json) => json.components as Component[])

    const scanners = await fetch(
      `${$config.serviceBaseUrl}/v1/scanners?fields=scanners(id,description)`
    )
      .then((res) => res.json())
      .then((json) => json.scanners as Scanner[])

    const tests = await fetch(
      `${$config.serviceBaseUrl}/v1/tests?fields=tests(id,description)`
    )
      .then((res) => res.json())
      .then((json) => json.tests as Test[])

    return {
      areas,
      teams,
      components,
      scanners,
      tests,
    }
  },
  data() {
    return {
      introTitle: this.$config.introTitle as String | undefined,
      introMarkdown: this.$config.introMarkdown as String | undefined,
      areas: [] as Area[],
      teams: [] as Team[],
      components: [] as Component[],
      scanners: [] as Scanner[],
      tests: [] as Test[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle',
    }
  },
  computed: {
    areaCount(): number {
      return this.areas.length
    },
    teamCount(): number {
      return this.teams.length
    },
    componentCount(): number {
      return this.components.length
    },
    componentTypeCounts(): ItemCount<string>[] {
      return itemCounts(this.components.map((component) => component.typeId))
    },
    platformCounts(): ItemCount<string>[] {
      return itemCounts(
        this.components.map((component) => component.platformId)
      )
    },
    tagCounts(): ItemCount<string>[] {
      return itemCounts(
        this.components.flatMap((component) => component.tags ?? [])
      )
    },
    techDebtCount(): number {
      return this.components.flatMap((component) => component.techDebts ?? [])
        .length
    },
    testResultCount(): number {
      return this.components.flatMap((component) => component.testResults ?? [])
        .length
    },
  },
})
</script>
