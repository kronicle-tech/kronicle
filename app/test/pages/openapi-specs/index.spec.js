import Index from '~/pages/openapi-specs'
import { createPageWrapper } from '~/test/pages/pageUtils'
import {
  createComponent,
  createComponentWithOpenApiSpecs,
} from '~/test/testDataUtils'

describe('Index', () => {
  let components = []
  let wrapper
  async function createWrapper() {
    wrapper = await createPageWrapper(Index, {
      serviceRequests: {
        '/v1/components?stateType=openapi-specs&fields=components(id,name,type,tags,teams,platformId,states)':
          {
            responseBody: { components },
          },
      },
    })
  }

  beforeEach(() => {
    components = []
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  test('has the right page title', async () => {
    await createWrapper()
    expect(wrapper.vm.$metaInfo.title).toBe('Kronicle - OpenAPI Specs')
  })

  describe('when Get Components service endpoint returns an empty array', () => {
    test('renders the page', async () => {
      await createWrapper()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('when Get Components service endpoint returns an array of multiple components', () => {
    beforeEach(() => {
      components = [
        createComponentWithOpenApiSpecs({ componentNumber: 1 }),
        createComponent({ componentNumber: 2 }),
        createComponentWithOpenApiSpecs({ componentNumber: 3 }),
      ]
    })

    test('renders the page', async () => {
      await createWrapper()
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
