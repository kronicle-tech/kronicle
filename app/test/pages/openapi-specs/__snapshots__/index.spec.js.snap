// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index when Get Components service endpoint returns an array of multiple components renders the page 1`] = `
<div
  class="m-3"
>
  <div>
    <div
      class="card my-3"
    >
      <!---->
      <!---->
      <div
        class="list-group"
      >
        <div
          class="list-group-item list-group-item-success"
        >
          <span
            class=""
          >
            4
          </span>
          
        OpenAPI specs
      
        </div>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="mt-3 mb-4"
      components="[object Object],[object Object],[object Object]"
    >
       
      <button
        aria-controls="filters"
        aria-expanded="false"
        class="btn btn-primary collapsed"
        id="toggleFilters"
        type="button"
      >
        <svg
          aria-hidden="true"
          aria-label="blank"
          class="bi-blank b-icon bi"
          fill="currentColor"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g />
        </svg>
         Filters
  
      </button>
       
      <transition-stub
        class="mt-3"
        css="true"
        enteractiveclass="collapsing"
        enterclass=""
        entertoclass="collapse show"
        leaveactiveclass="collapsing"
        leaveclass="collapse show"
        leavetoclass="collapse"
      >
        <div
          class="collapse"
          id="filters"
          style="display: none;"
        >
          <div
            class="card-columns"
          >
             
            <!---->
             
            <!---->
             
            <!---->
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__24"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__24__BV_label_"
                    tabindex="-1"
                  >
                    Teams
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__25"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__25_BV_option_0"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__25_BV_option_0"
                        >
                          <span>
                            test-team-id-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__25_BV_option_1"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__25_BV_option_1"
                        >
                          <span>
                            test-team-id-1-2
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__25_BV_option_2"
                          name="team"
                          type="checkbox"
                          value="test-team-id-2-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__25_BV_option_2"
                        >
                          <span>
                            test-team-id-2-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__25_BV_option_3"
                          name="team"
                          type="checkbox"
                          value="test-team-id-2-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__25_BV_option_3"
                        >
                          <span>
                            test-team-id-2-2
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__25_BV_option_4"
                          name="team"
                          type="checkbox"
                          value="test-team-id-3-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__25_BV_option_4"
                        >
                          <span>
                            test-team-id-3-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__25_BV_option_5"
                          name="team"
                          type="checkbox"
                          value="test-team-id-3-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__25_BV_option_5"
                        >
                          <span>
                            test-team-id-3-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__32"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__32__BV_label_"
                    tabindex="-1"
                  >
                    Tags
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__33"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__33_BV_option_0"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__33_BV_option_0"
                        >
                          <span>
                            test-tag-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__33_BV_option_1"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__33_BV_option_1"
                        >
                          <span>
                            test-tag-1-2
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__33_BV_option_2"
                          name="tag"
                          type="checkbox"
                          value="test-tag-2-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__33_BV_option_2"
                        >
                          <span>
                            test-tag-2-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__33_BV_option_3"
                          name="tag"
                          type="checkbox"
                          value="test-tag-2-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__33_BV_option_3"
                        >
                          <span>
                            test-tag-2-2
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__33_BV_option_4"
                          name="tag"
                          type="checkbox"
                          value="test-tag-3-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__33_BV_option_4"
                        >
                          <span>
                            test-tag-3-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__33_BV_option_5"
                          name="tag"
                          type="checkbox"
                          value="test-tag-3-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__33_BV_option_5"
                        >
                          <span>
                            test-tag-3-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__40"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__40__BV_label_"
                    tabindex="-1"
                  >
                    Component Types
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__41"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__41_BV_option_0"
                          name="componentType"
                          type="checkbox"
                          value="test-component-type-id-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__41_BV_option_0"
                        >
                          <span>
                            test-component-type-id-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__41_BV_option_1"
                          name="componentType"
                          type="checkbox"
                          value="test-component-type-id-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__41_BV_option_1"
                        >
                          <span>
                            test-component-type-id-2
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__41_BV_option_2"
                          name="componentType"
                          type="checkbox"
                          value="test-component-type-id-3"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__41_BV_option_2"
                        >
                          <span>
                            test-component-type-id-3
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__45"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__45__BV_label_"
                    tabindex="-1"
                  >
                    Platforms
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__46"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__46_BV_option_0"
                          name="platformId"
                          type="checkbox"
                          value="undefined"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__46_BV_option_0"
                        >
                          <span>
                            undefined
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <div
                  class="form-group"
                  id="__BVID__48"
                  role="group"
                >
                  <label
                    class="d-block"
                    for="component-filter"
                    id="__BVID__48__BV_label_"
                  >
                    Component
                  </label>
                  <div>
                    <select
                      class="custom-select custom-select-sm"
                      id="component-filter"
                    >
                      <option
                        value=""
                      >
                        Please select a component
                      </option>
                      <option
                        value="test-component-id-1"
                      >
                        test-component-id-1
                      </option>
                      <option
                        value="test-component-id-2"
                      >
                        test-component-id-2
                      </option>
                      <option
                        value="test-component-id-3"
                      >
                        test-component-id-3
                      </option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
     
    <table
      class="table table-dark"
    >
      <thead>
        <tr>
          <th
            class="component"
          >
            Component
          </th>
           
          <th
            class="teams"
          >
            Teams
          </th>
           
          <th
            class="action"
          >
            Action
          </th>
           
          <th
            class="location"
          >
            Location
          </th>
           
          <th
            class="description"
          >
            Description
          </th>
        </tr>
      </thead>
       
      <tbody>
        <tr>
          <td
            class="component"
          >
            <a
              href="/components/test-component-id-1"
            >
              Test Component Name 1
            </a>
          </td>
           
          <td
            class="teams"
          >
            <ul>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-1-1"
                >
                  test-team-id-1-1
                </a>
                 
                <!---->
                 
                <!---->
              </li>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-1-2"
                >
                  test-team-id-1-2
                </a>
                 
                <!---->
                 
                <!---->
              </li>
            </ul>
          </td>
           
          <td
            class="action table-secondary"
          >
            <div>
              <span
                class="badge badge-danger"
              >
                OpenAPI spec not found
              </span>
            </div>
          </td>
           
          <td
            class="location"
          >
            
        https://example.com/test-openapi-spec-1-1
      
          </td>
           
          <td
            class="description"
          >
            <div>
              

              <p>
                Test OpenAPI Spec Description 1 1
              </p>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="component"
          >
            <a
              href="/components/test-component-id-1"
            >
              Test Component Name 1
            </a>
          </td>
           
          <td
            class="teams"
          >
            <ul>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-1-1"
                >
                  test-team-id-1-1
                </a>
                 
                <!---->
                 
                <!---->
              </li>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-1-2"
                >
                  test-team-id-1-2
                </a>
                 
                <!---->
                 
                <!---->
              </li>
            </ul>
          </td>
           
          <td
            class="action table-secondary"
          >
            <div>
              <span
                class="badge badge-danger"
              >
                OpenAPI spec not found
              </span>
            </div>
          </td>
           
          <td
            class="location"
          >
            
        https://example.com/test-openapi-spec-1-2
      
          </td>
           
          <td
            class="description"
          >
            <div>
              

              <p>
                Test OpenAPI Spec Description 1 2
              </p>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="component"
          >
            <a
              href="/components/test-component-id-3"
            >
              Test Component Name 3
            </a>
          </td>
           
          <td
            class="teams"
          >
            <ul>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-3-1"
                >
                  test-team-id-3-1
                </a>
                 
                <!---->
                 
                <!---->
              </li>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-3-2"
                >
                  test-team-id-3-2
                </a>
                 
                <!---->
                 
                <!---->
              </li>
            </ul>
          </td>
           
          <td
            class="action table-secondary"
          >
            <div>
              <span
                class="badge badge-danger"
              >
                OpenAPI spec not found
              </span>
            </div>
          </td>
           
          <td
            class="location"
          >
            
        https://example.com/test-openapi-spec-3-1
      
          </td>
           
          <td
            class="description"
          >
            <div>
              

              <p>
                Test OpenAPI Spec Description 3 1
              </p>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="component"
          >
            <a
              href="/components/test-component-id-3"
            >
              Test Component Name 3
            </a>
          </td>
           
          <td
            class="teams"
          >
            <ul>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-3-1"
                >
                  test-team-id-3-1
                </a>
                 
                <!---->
                 
                <!---->
              </li>
              <li>
                <a
                  class="team-name"
                  href="/teams/test-team-id-3-2"
                >
                  test-team-id-3-2
                </a>
                 
                <!---->
                 
                <!---->
              </li>
            </ul>
          </td>
           
          <td
            class="action table-secondary"
          >
            <div>
              <span
                class="badge badge-danger"
              >
                OpenAPI spec not found
              </span>
            </div>
          </td>
           
          <td
            class="location"
          >
            
        https://example.com/test-openapi-spec-3-2
      
          </td>
           
          <td
            class="description"
          >
            <div>
              

              <p>
                Test OpenAPI Spec Description 3 2
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Index when Get Components service endpoint returns an empty array renders the page 1`] = `
<div
  class="m-3"
>
  <div>
    <div
      class="card my-3"
    >
      <!---->
      <!---->
      <div
        class="list-group"
      >
        <div
          class="list-group-item list-group-item-danger"
        >
          <span
            class="display-1"
          >
            0
          </span>
          
        OpenAPI specs
      
        </div>
      </div>
      <!---->
      <!---->
    </div>
     
    <!---->
     
    <!---->
  </div>
</div>
`;
